solutions/usaco-944.mdx

```diff
@@ -0,0 +1,103 @@
+---
+id: usaco-944
+source: USACO Silver 2019 US Open
+title: Fence Planning
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_fenceplan_silver_open19.html)
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+  
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class FencePlanning {
+	static Cow[] cows;
+	static List<Integer>[] graph;
+	static boolean[] visited;
+	static int lowX = Integer.MAX_VALUE;
+	static int highX = Integer.MIN_VALUE;
+	static int lowY = Integer.MAX_VALUE;
+	static int highY = Integer.MIN_VALUE;
+
+	static class Cow { // class to keep track of the x and y
+		int x;
+		int y;
+	}
+
+	static void floodfill (int currentCow) { // dfs flood fill
```

Comment:
~~ew recursion~~
honestly an iterative dfs would be better but who am i to judge

Reply:
recursive is simpler and it passes the problem so